### ISSUE TITLE
head/tail/top error messages should speak of constant expressions

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#5297]
## tl;dr

This error message speaks of an expression/integer, but it actually takes a compile-time constant expression that evaluates to a positive integer, which is more limiting.

```
$ zq 'yield {x:1} | head x'
zq: expression value must be an integer value: error("missing") at line 1, column 20:
yield {x:1} | head x
                   ~
```

## Details

Repro is with Zed commit c1da204.

This topic was surfaced in a recent [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1727020579850509). Similar error messages show up with `head`, `tail`, and `top`.

```
$ zq -version
Version: v1.18.0

$ zq 'yield {x:1} | head x'
zq: expression value must be an integer value: error("missing") at line 1, column 20:
yield {x:1} | head x
                   ~

$ zq 'yield {x:1} | tail x'
zq: expression value must be an integer value: error("missing") at line 1, column 20:
yield {x:1} | tail x
                   ~

$ zq 'yield {x:1} | top x x'
zq: limit argument must be an integer at line 1, column 19:
yield {x:1} | top x x
                  ~
```

In recent PRs #5276 and #5296 we're making sure user-facing docs emphasize that each of these parameters should be a "compile-time constant expression that evaluates to a positive integer". However, since some users may not have read the docs, we could stand to improve the error message.

**Comment by philrz** [comment]
Verified in super commit 0165f09.

Repeating the repros with modern syntax and each of the operators mentioned above, the error messages now speak of constant expressions.

```
$ super -version
Version: 0165f094e

$ super -c 'values {x:1} | head x'
cannot reference 'x' in constant expression at line 1, column 21:
values {x:1} | head x
                    ~

$ super -c 'values {x:1} | tail x'
cannot reference 'x' in constant expression at line 1, column 21:
values {x:1} | tail x
                    ~

$ super -c 'values {x:1} | top x x'
cannot reference 'x' in constant expression at line 1, column 20:
values {x:1} | top x x
                   ~
```

Thanks @mccanne!
